tests/unit_tests/db_engine_specs/test_kusto.py

```diff
@@ -0,0 +1,104 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=unused-argument, import-outside-toplevel, protected-access
+import pytest
+from flask.ctx import AppContext
+
+
+@pytest.mark.parametrize(
+    "sql,expected",
+    [
+        ("SELECT foo FROM tbl", True),
+        ("SHOW TABLES", False),
+        ("EXPLAIN SELECT foo FROM tbl", False),
+        ("INSERT INTO tbl (foo) VALUES (1)", False),
+    ],
+)
+def test_sql_is_readonly_query(
+    app_context: AppContext, sql: str, expected: bool
+) -> None:
+    """
+    Make sure that SQL dialect consider only SELECT statements as read-only
+    """
+
+    from superset.db_engine_specs.kusto import KustoSqlEngineSpec
+    from superset.sql_parse import ParsedQuery
+
+    parsed_query = ParsedQuery(sql)
+    is_readonly = KustoSqlEngineSpec.is_readonly_query(parsed_query)
+
+    assert expected == is_readonly
+
+
+@pytest.mark.parametrize(
+    "kql,expected",
+    [
+        ("tbl | limit 100", True),
+        ("let foo = 1; tbl | where bar == foo", True),
+        (".show tables", False),
+    ],
+)
+def test_kql_is_select_query(app_context: AppContext, kql: str, expected: bool) -> None:
+    """
+    Make sure that KQL dialect consider only statements that do not start with "." (dot)
+    as a SELECT statements
+    """
+
+    from superset.db_engine_specs.kusto import KustoKqlEngineSpec
+    from superset.sql_parse import ParsedQuery
+
+    parsed_query = ParsedQuery(kql)
+    is_select = KustoKqlEngineSpec.is_readonly_query(parsed_query)
+
+    assert expected == is_select
+
+
+@pytest.mark.parametrize(
+    "kql,expected",
+    [
+        ("tbl | limit 100", True),
+        ("let foo = 1; tbl | where bar == foo", True),
+        (".show tables", False),
```

Comment:
Again, out of curiosity, is this actually a read-only query? I know that `SHOW TABLES`, `EXPLAIN` etc appears to currently be flagged as DML by `parse_sql`, but as this logic will now contained in the db engine spec, this can now be easily altered. So maybe we'd like to change this to be flagged as read-only in the KQL spec? Also, could we add a more typical DML case here, like `.set`, `.append` or `.drop`? I assume they're the cases we want to make sure we're catching.

Reply:
Your remark is valuable. Thank you. We will add more specific code.